dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -2872,6 +2933,49 @@ Sedp::Writer::set_header_fields(DCPS::DataSampleHeader& dsh,
   dsh.source_timestamp_nanosec_ = now.usec() * 1000;
 }
 
+#ifdef OPENDDS_SECURITY
+Sedp::RepeatOnceWriter::RepeatOnceWriter(const RepoId& pub_id, Sedp& sedp)
+  : Writer(pub_id, sedp)
+{
+}
+
+void
+Sedp::RepeatOnceWriter::acknowledged_by_reader(const RepoId& rdr)
+{
+  RepoId remote_participant;
+  RTPS::assign(remote_participant.guidPrefix, rdr.guidPrefix);
+  remote_participant.entityId = ENTITYID_PARTICIPANT;
+
+  if (!std::memcmp(repo_id_.guidPrefix, rdr.guidPrefix, sizeof(DCPS::GuidPrefix_t))) {
+    return;
+  }
+
+  if (DCPS::DCPS_debug_level > 3) {
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) Sedp::RepeatOnceWriter::acknowledged_by_reader(): %C\n",
+      OPENDDS_STRING(DCPS::GuidConverter(rdr)).c_str()));
+  }
+
+  ACE_Guard<ACE_Thread_Mutex> g(lock_);
+  if (remote_readers_.count(remote_participant)) {
+    RemoteReaderVector& remote_readers = remote_readers_[remote_participant];
```

Comment:
This does the lookup twice (count + []). Use find() to grab the iterator, check to see it's not equal to .end(), and then dereference it get the value. Same applies for line 2969.

Reply:
3 times, if you count erase!